project/Release.scala

```diff
@@ -0,0 +1,69 @@
+import sbt.Keys._
+import sbt.{Command, _}
+
+import scala.sys.process
+
+object Release {
+
+  // Main release commands
+  private val stryker4sPublish = "stryker4sPublish"
+  private val stryker4sPublishSigned = "stryker4sPublishSigned"
+  // Helper command names
+  private val stryker4sMvnSetup = "stryker4sMvnSetup"
+  private val stryker4sMvnPackage = "stryker4sMvnPackage"
+  private val stryker4sMvnDeploy = "stryker4sMvnDeploy"
+  private val publishM2 = "stryker4s-core/publishM2"
+  private val `+publish` = "+publish"
+  private val `+publishSigned` = "+publishSigned"
+
+  lazy val releaseCommands: Setting[Seq[Command]] = {
+    commands ++= {
+      val versionNumber = version.value // Remember the version because sbt-dynver will try to set another one after changes in runner/maven/pom.xml
+      Seq(
+        // Called by sbt-ci-release
+        Command.command(stryker4sPublish)(stryker4sMvnSetup :: `+publish` :: stryker4sMvnDeploy :: _),
+        Command.command(stryker4sPublishSigned)(stryker4sMvnSetup :: `+publishSigned` :: stryker4sMvnDeploy :: _),
+        // Called by stryker4sPublish(signed)
+        Command.command(stryker4sMvnSetup)(publishM2 :: stryker4sMvnPackage :: setVersion(versionNumber) :: _),
+        Command.command(stryker4sMvnPackage)(mvnPackage(versionNumber, baseDirectory.value)),
+        Command.command(stryker4sMvnDeploy)(mvnDeploy(baseDirectory.value))
+      )
+    }
+  }
+
+  /** First publish `stryker4s-core` to local '''maven''' repository,
+    * then set new version and run `mvn install`
+    */
+  private def mvnPackage(version: String, baseDir: File): State => State = handleProcessResult(
+    _,
+    mvnGoal(s"versions:set -DnewVersion=$version", baseDir) #&&
+      mvnGoal("package", baseDir)
+  )
+
+  /** Deploy the maven plugin project
+    */
```

Comment:
Docs like these feel a bit redundant. This is basically a more verbose version of the function name.

I don't really understand why there's so many scaladoc here. 🤔

Reply:
Yeah I went a bit scaladoc-crazy. It looked more complex at first.